MovieQuiz/QuestionFactory.swift

```diff
@@ -8,62 +8,109 @@
-            image: "Deadpool",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: true),
-        QuizQuestion(
-            image: "The Green Knight",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: true),
-        QuizQuestion(
-            image: "Old",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false),
-        QuizQuestion(
-            image: "The Ice Age Adventures of Buck Wild",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false),
-        QuizQuestion(
-            image: "Tesla",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false),
-        QuizQuestion(
-            image: "Vivarium",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false)
-    ]
-    
+//    private let questions: [QuizQuestion] = [
+//        QuizQuestion(
+//            image: "The Godfather",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: true),
+//        QuizQuestion(
+//            image: "The Dark Knight",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: true),
+//        QuizQuestion(
+//            image: "Kill Bill",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: true),
+//        QuizQuestion(
+//            image: "The Avengers",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: true),
+//        QuizQuestion(
+//            image: "Deadpool",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: true),
+//        QuizQuestion(
+//            image: "The Green Knight",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: true),
+//        QuizQuestion(
+//            image: "Old",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: false),
+//        QuizQuestion(
+//            image: "The Ice Age Adventures of Buck Wild",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: false),
+//        QuizQuestion(
+//            image: "Tesla",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: false),
+//        QuizQuestion(
+//            image: "Vivarium",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: false)
+//    ]
+    private var movies: [MostPopularMovie] = []
+    private let moviesLoader: MoviesLoading
     weak var delegate: QuestionFactoryDelegate?
     
-    init(delegate: QuestionFactoryDelegate?) {
+    init(delegate: QuestionFactoryDelegate, moviesLoader: MoviesLoading) {
         self.delegate = delegate
+        self.moviesLoader = moviesLoader
+    }
+    
+    func loadData() {
+        moviesLoader.loadMovies { result in
```

Comment:
Здесь не приводит к ошибке и retain cycle, но формально лучше ослабить ссылку в верхнем замыкании. 
```suggestion
        moviesLoader.loadMovies { [weak self] result in
```